src/plugins/intel_npu/src/al/include/intel_npu/al/icompiler.hpp

```diff
@@ -87,13 +87,39 @@ struct NetworkDescription final {
     NetworkMetadata metadata;
 };
 
+/**
+ * @struct Version
+ * @brief A helper structure used to validate compiler versions.
+ */
+struct Version {
+    uint32_t major;
+    uint32_t minor;
+    uint32_t patch;
+
+    inline bool isCompatible(const Version& other) const {
+        return this->major == other.major && this->minor == other.minor;  // TODO: needs confirmation
```

Comment:
Just a reminder to remove this comment upon receiving the confirmation. Perhaps we should do this prior to merging the PR.

Reply:
What kind of confirmation do we need here? If we still need such a confirmation after merging the PR, could we open a ticket and remove this comment?

Reply:
We should be able to support backward compatibility by this check. So we need to support cases when driver minor version >= compiler minor version

Reply:
updated the check and moved it inside plugin.cpp

Reply:
Sorry, I do not see the check inside plugin.cpp 

https://github.com/openvinotoolkit/openvino/pull/24502/files#diff-f1f8f14a46f6bfbbfc9561e65e1a27b99d0e3a554964137190d414557eec728cR549-R572

I see that we checked that versions are not nullpts and then adding them into config. Am I missing something?

Reply:
The check was moved in the compiler: https://github.com/openvinotoolkit/openvino/pull/24502/files#diff-f1f8f14a46f6bfbbfc9561e65e1a27b99d0e3a554964137190d414557eec728cR589 . My bad, I made a typo in the comment.